src/read/elf/file.rs

```diff
@@ -377,10 +377,11 @@ where
         let mut notes = self.notes();
         while let Some(mut notes) = notes.next().transpose()? {
             while let Some(note) = notes.next()? {
-                if note.name() == elf::ELF_NOTE_GNU
-                    && note.n_type(self.endian) == elf::NT_GNU_BUILD_ID
-                {
-                    return Ok(Some(note.desc()));
+                match (note.name(), note.n_type(self.endian)) {
```

Comment:
Please leave it as the if statement because it nests less.

Reply:
Done.